machineconfiguration/v1alpha1/types_machineconfignode.go

```diff
@@ -155,6 +190,20 @@ type MachineConfigNodeSpecMachineConfigVersion struct {
 	Desired string `json:"desired"`
 }
 
+type MachineConfigNodeSpecPinnedImageSetVersion struct {
+	// name is the name of the pinned image set.
+	// Must be a lowercase RFC-1123 hostname (https://tools.ietf.org/html/rfc1123)
+	// It may consist of only alphanumeric characters, hyphens (-) and periods (.)
+	// and must be at most 253 characters in length.
+	// +kubebuilder:validation:MaxLength:=253
+	// +kubebuilder:validation:Pattern=`^([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]{0,61}[a-zA-Z0-9])(\.([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]{0,61}[a-zA-Z0-9]))*$`
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+	// desiredGeneration is the generation of the pinned image set that should be pinned and pulled on this node.
+	// +kubebuilder:validation:Required
+	DesiredGeneration int32 `json:"desiredGeneration"`
```

Comment:
This construct is a little strange to me. I understand trying to set which pinnedimageset to use (maybe), but I wouldn't expect to see the generation codified since anything other than "current" is impossible to read from clusterstate.  On status this makes sense so you can see "I have a diff", but I don't understand how a desiredgeneration less than the current generation makes sense on spec.

Reply:
yeah I dropped it and it allows for MCP spec to reflect 1:1